content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,430 @@
+
+#### <inlinecode>OR</inlinecode> operators
+
+The following code in Prisma 4 and lower:
+
+```js
+prisma.user.findMany({
+  OR: { email: 'foo@example.com' },
+})
+```
+
+Will need to be changed to the following in Prisma 5:
+
+```js highlight=2;edit
+prisma.user.findMany({
+  OR: [{ email: 'foo@example.com' }],
+})
+```
+
+`OR` operators will only accept array values.
+
+#### <inlinecode>in</inlinecode> and <inlinecode>notIn</inlinecode> operators
+
+Similar to `OR`, `in` and `notIn` require array values.
+
+Prisma 4 and lower:
+
+```js
+prisma.user.findMany({
+  id: { in: 123 },
+})
+
+prisma.user.findMany({
+  id: { notIn: 123 },
+})
+```
+
+Prisma 5:
+
+```js highlight=3,9;edit
+prisma.user.findMany({
+  id: {
+    in: [123],
+  },
+})
+
+prisma.user.findMany({
+  id: {
+    notIn: [123],
+  },
+})
+```
+
+If your `in` and `notIn` values are only one element, you can also update your code to not use these operators:
+
+```js highlight=2,6;normal
+prisma.user.findMany({
+  id: 123,
+})
+
+prisma.user.findMany({
+  id: { not: 123 },
+})
+```
+
+#### <inlinecode>path</inlinecode> argument for filtering on JSON fields in Postgres
+
+[When filtering on JSON fields in a Postgres table](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields#filter-on-a-json-field) the `path` argument now only accepts an array.
+
+When using the following schema:
+
+```prisma
+datasource db {
+  provider = "postgresql"
+}
+
+model User {
+  id       String @id
+  settings Json
+}
+```
+
+Prisma 4 and lower:
+
+```js
+prisma.user.findMany({
+  where: {
+    settings: {
+      path: 'someSetting',
+      equals: someValue,
+    },
+  },
+})
+```
+
+Prisma 5:
+
+```js highlight=4;edit
+prisma.user.findMany({
+  where: {
+    settings: {
+      path: ['someSetting'],
+      equals: someValue,
+    },
+  },
+})
+```
+
+<Admonition type="info">
+
+Note: This change only affects PostgreSQL databases. MySQL databases are not affected.
+
+</Admonition>
+
+#### Composite and Scalar lists
+
+Inputs for operations on lists of [Composite](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#defining-composite-types) and [Scalar](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#scalar-fields) types now only accept array values. The following examples all use the following schema:
+
+```prisma
+model Post {
+  id String @id @default(uuid())
+  commentsList Comment[]
+  tags String[]
+}
+
+type Comment {
+  text String
+}
+```
+
+##### Composite list filtering
+
+[When filtering records based on a composite list field](https://www.prisma.io/docs/concepts/components/prisma-client/composite-types#finding-records-that-contain-composite-types-with-find-and-findmany) you will only be able to supply array values.
+
+Prisma 4 and lower:
+
+```js
+prisma.post.findMany({
+  commentsList: { equals: { text: 'hello' } },
+})
+```
+
+Prisma 5:
+
+```js highlight=2-4;normal
```

Comment:
I noticed that there was a visual difference between `js highlight=2-4;normal` and `js highlight=4;edit` (line 139) - is it intentional that we use a different highlighting mode? (I think normal looks/feels better, I was a little confused why I "have a line selected" (or it looked like it) in the blocks using `edit` mode highlights)